pkg/ruler/remotequerier.go

```diff
@@ -266,16 +266,20 @@ func (q *RemoteQuerier) sendRequest(ctx context.Context, req *httpgrpc.HTTPReque
 	}
 	retry := backoff.New(ctx, retryConfig)
 
-	for retry.Ongoing() {
+	for {
 		resp, err := q.client.Handle(ctx, req)
-		if err != nil {
+		if err == nil {
+			return resp, nil
+		}
+		retry.Wait()
+
+		if retry.Ongoing() {
 			level.Warn(q.logger).Log("msg", "failed to remotely evaluate query expression, will retry", "err", err)
```

Comment:
This log should happen when the request has failed, not after waiting. Otherwise it will be complicated to correlate the logs (_I see the failed frontend request at 9:00:00 but I see the log at 9:00:02_)

Reply:
nice catch. makes perfect sense.

changed @ 6920b72878a298c7052647556bc7773e47c0f75a